### ISSUE TITLE
Unsupported CPU model 85

### ISSUE THREAD

**Issue by nolanw123** [andikleen/pmu-tools#247]
Hi,

After a fresh git clone I ran toplev.py and got:

```
bob@instance-1 ~ $ toplev.py
Unsupported CPU model 85
```

This looks like it might be a repeat of issue #177 from 2019.
However, the code was sufficiently different from back then that I wasn't able to discern what exactly was going wrong.

Here's my /proc/cpuinfo:

```
bob@instance-1 ~ $ cat /proc/cpuinfo | head -26
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 85
model name      : Intel(R) Xeon(R) CPU @ 2.00GHz
stepping        : 3
microcode       : 0x1
cpu MHz         : 2000.000
cache size      : 39424 KB
physical id     : 0
siblings        : 16
core id         : 0
cpu cores       : 8
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc eagerfpu pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 arat md_clear spec_ctrl intel_stibp arch_capabilities
bogomips        : 4000.00
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
```

Please let me know if I can supply any more information that might help with debugging.

**Comment by andikleen** [comment]
> Here's my /proc/cpuinfo:
> 
> bob@instance-1 ~ $ cat /proc/cpuinfo | head -26
> processor       : 0
> vendor_id       : GenuineIntel
> cpu family      : 6
> model           : 85
> model name      : Intel(R) Xeon(R) CPU @ 2.00GHz
> stepping        : 3

Is this a production system?

toplev was assuming SkylakeX has at least stepping 4

This would fix it:

diff --git a/toplev.py b/toplev.py
index 331def3..e802f5b 100755
--- a/toplev.py
+++ b/toplev.py
@@ -45,6 +45,7 @@ known_cpus = (
     ("skl", (94, 78, 142, 158, )),
     ("knl", (87, )),
     ("skx", ((85, 4,), )),
+    ("skx", ((85, 3,), )),
     ("clx", ((85, 5,), )),
     ("icl", (126, )),
 )

But I would prefer to only support production systems.

**Comment by nolanw123** [comment]
Fair enough, it's not a prod system.  I'll close given it hasn't come up.  Thanks.

**Comment by andikleen** [comment]
On Mon, Mar 23, 2020 at 09:59:35PM -0700, nolanw123 wrote:
> Closed #247.

Well you can apply the patch, then it should work. I'll probably not
include it by default tho.
